### DISCUSSION TITLE
Error when using env variable from secrets

### DISCUSSION THREAD

**Discussion by claudiunicolaa** [community/community#25845] [Actions]
<p>Hello,</p>
<p>I tried to generate a file and push it to the repository, but I receive the error:<em>syntax error: unexpected end of file</em> whenI try to set the user.email.</p>
<p>The action configuration:</p>
<pre><code>name: generate-index

on:
  push:
    branches:
      - master
    paths:
      - resume.json

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v1
    - uses: actions/npm@master

    - name: Install jsonresume and generate
      run: |
        npm install resume-cli
        npm install jsonresume-theme-classy
        ./node_modules/.bin/resume export index.html --theme classy

    - name: Git push
      env:
        GIT_OWNER_EMAIL: ${{ secrets.GIT_OWNER_EMAIL }}
        GIT_OWNER_NAME: ${{ secrets.GIT_OWNER_NAME }}
        GIT_REMOTE_URL: ${{ secrets.GIT_REMOTE_URL }}
      run: |
        git config user.email "$GIT_OWNER_EMAIL"
        git config user.name "$GIT_OWNER_NAME"
        if [[`git status --porcelain`]]; then
          git add index.html
          git commit -a -m 'Updated index'
          git remote rm origin
          git remote add origin $GIT_REMOTE_URL
          git push origin HEAD:master
</code></pre>
<p>Secrets panel:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/c/c70a6ab98eafdfb063b0e589b0b9793187af99d8.png" data-download-href="/uploads/short-url/soNkKD8DAOIcN9xhzzoSarYsiY8.png?dl=1" title="ss.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/15197275/181093704-14d93932-eac7-46fb-8f44-5f0d636435c9.png" alt="ss.png" data-base62-sha1="soNkKD8DAOIcN9xhzzoSarYsiY8" width="690" height="398" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/c/c70a6ab98eafdfb063b0e589b0b9793187af99d8_2_690x398.png, https://aws1.discourse-cdn.com/github/optimized/2X/c/c70a6ab98eafdfb063b0e589b0b9793187af99d8_2_1035x597.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/c/c70a6ab98eafdfb063b0e589b0b9793187af99d8.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/c/c70a6ab98eafdfb063b0e589b0b9793187af99d8_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">ss.png</span><span class="informations">1078×622 40 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>The action’s output:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/2e266c50afe40f20550233765fb24942a0e80f6a.png" data-download-href="/uploads/short-url/6AghEDI78CEDgaRZ1WkVTFCQI7E.png?dl=1" title="ss1.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/15197275/181093707-02978abd-726b-4ed3-aaf6-dd4d764e4dbd.png" alt="ss1.png" data-base62-sha1="6AghEDI78CEDgaRZ1WkVTFCQI7E" width="690" height="188" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/2/2e266c50afe40f20550233765fb24942a0e80f6a_2_690x188.png, https://aws1.discourse-cdn.com/github/optimized/2X/2/2e266c50afe40f20550233765fb24942a0e80f6a_2_1035x282.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/2/2e266c50afe40f20550233765fb24942a0e80f6a_2_1380x376.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/2e266c50afe40f20550233765fb24942a0e80f6a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">ss1.png</span><span class="informations">1581×431 27 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Also, here it is the link to the repository: <a href="https://github.com/claudiunicolaa/claudiunicola.xyz" rel="nofollow noopener">https://github.com/claudiunicolaa/claudiunicola.xyz</a></p>

**Comment by wipe2238** [comment, marked as answer]
<p>You forgot “fi” :wink:</p>

**Comment by claudiunicolaa** [comment]
<p>Well, I tried a lot of things but nothing regarding the “if” statement. :))</p>
<p>Thanks.</p>
